tictactoe_engine.py

```diff
@@ -10,82 +11,77 @@ class Code(Enum):
 	VALID = 6
 
 class tictactoeEngine:
-	def __init__(self, agent1, agent1moves, agent1value,  agent2, agent2moves, agent2value):
+	def __init__(self, agent1, agent2):
 		self.agent1 = agent1
 		self.agent2 = agent2
-		self.agent1moves = agent1moves
-		self.agent2moves = agent2moves
-		self.agent1value = agent1value
-		self.agent2value = agent2value
 		self.gamestate = [0] * 9
-		self.index1 = 0
-		self.index2 = 0
+		self.state_list= []
 
+	
 	def startEngine(self):
 		if (self.gamestate == [0] * 9):
-			if (self.checkWin() == Code.RUNNING):
+			if (self.checkWin() == engineReturnCode.RUNNING):
 				for x in range(0,9):
-					if(x%2 == 0 and (self.checkTie() != Code.TIE) and (self.checkWin()== Code.RUNNING) ):
-						a = self.agent1moves[self.index1]
-						while(self.checkvalid(a-1) != Code.SUCCESS):
-							a = self.agent1moves[self.index1]
-						self.index1 +=1
-						self.gamestate[a-1] = self.agent1value
-					elif(x%2==1 and (self.checkTie() != Code.TIE) and (self.checkWin()== Code.RUNNING) ):
-						b = self.agent2moves[self.index2]
-						while(self.checkvalid(b-1) != Code.SUCCESS):
-							b = self.agent2moves[self.index2]
-						self.index2 +=1
-						self.gamestate[b-1] = self.agent2value
+					if(x%2 == 0 and (self.checkTie() != engineReturnCode.TIE) and (self.checkWin()== engineReturnCode.RUNNING) ):
+						a = self.agent1.move(self.gamestate)
+						while(self.checkvalid(a) != engineReturnCode.SUCCESS):
+							a = self.agent1.move(self.gamestate)
+						self.gamestate[a] = self.agent1.value
+					elif(x%2==1 and (self.checkTie() != engineReturnCode.TIE) and (self.checkWin()== engineReturnCode.RUNNING) ):
+						b = self.agent2.move(self.gamestate)
+						while(self.checkvalid(b) != engineReturnCode.SUCCESS):
+							b = (self.agent2).move(self.gamestate)
+						self.gamestate[b] = self.agent2.value
```

Comment:
- Game engine should not be controlling the agents.
- Game engine should not be affected/bothered by what functions agent has and how those functions should be called to make it play. _Implies: no calls to agent or its functions from game engine._

If it was not clear already, then _this function `startEngine` should not exist._ 
This game engine is an environment (not a controller) - it does not make anyone act. It only responds to actions on it. Accept an action, if action is valid then change the state else report the problem.
The action can come from agent, human etc. - source of action should not matter.

There should be two (simpler) functions instead:
For an agent / human / interface
- to get current board
- to play a move

_Carry over from last review:_

> Where is the function to make a single move? How can a player / agent / interface get the current board and make a move?